### PR TITLE
when showing targets, eliminate gaps in years between them and data point values

### PR DESCRIPTION
**Pull request by tsubik**
Could be tested on Indonesia platform.

Both line and area charts for National or Province GHG Emissions.

Before

image

After

image

### CODE REVIEW

**Comment by weberjavi** [comment]
on it ☝️

File: src/components/charts/selectors/chart-selectors.js [hunk 1]
```diff
@@ -48,13 +55,7 @@ const getDataMax = createSelector(getDataWithTotal, data => {
...
 
 export const getDomain = createSelector(
-  [
-    getData,
-    state => state.config,
-    state => state.projectedData,
-    getDataMin,
-    getDataMax
-  ],
+  [ getData, state => state.config, getProjectedData, getDataMin, getDataMax ],
```
Comment (weberjavi) [inline src/components/charts/selectors/chart-selectors.js]: Just for the sake of consistency maybe we could add another _getter_ for the config to use in the selector, as you've made with `getProjectedData`?
`getConfig = state => state.config || null`
`export const getDomain = createSelector([getData, getConfig, ...], (data, config, ...) = {...})`

Comment (tsubik) [reply src/components/charts/selectors/chart-selectors.js]: Ok! Good idea!

**Comment by tsubik** [comment]
I'm thinking now that I should give more options here. I was looking at GHG emission on the flagship, and when there are more years with records, bringing targets with an only a one-year interval to the last value will not look any good. On Indonesia it looks good, because there are only records for a few years, and then many years gap to the first target.

**Comment by tsubik** [comment]
I have to experiment with more xValues :)

**Comment by tsubik** [comment]
All right, fixed. Now it will make gaps between the last data value and target dynamic.

**Pull request merged**